server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1835,6 +1847,26 @@ private Volume orchestrateDetachVolumeFromVM(long vmId, long volumeId) {
         }
     }
 
+    public void updateMissingRootDiskController(final VMInstanceVO vm, final String rootVolChainInfo) {
+        if (vm == null || !vm.getType().equals(VirtualMachine.Type.User) || Strings.isNullOrEmpty(rootVolChainInfo)) {
```

Comment:
@bhaisaab I was no talking about removing the "vm != null", that check I know we need to execute. 
I was only refering to the "vm.getType().equals(VirtualMachine.Type.User)", the "vm.getType" can return a null value.

Did you understand what I mean? Just inverting the comparison as I said before.
from
`vm.getType().equals(VirtualMachine.Type.User)`
to 
`VirtualMachine.Type.User.equals(vm.getType())`